### ISSUE TITLE
Make persistence model a proper model that can be called

### ISSUE THREAD

**Issue by HansBambel** [HansBambel/SmaAt-UNet#28]
The persistence model is just manually selecting the last input. It requires a separate function to be calculated. It would be better if it is a proper model that can be called just like the other models in the testing scripts.

**Comment by HansBambel** [comment]
I am thinking that a model inherits from pl.Lightningmodule and implements the forward pass by just taking the last image of the input and returning that.

Something along those lines:
```python
class PersistenceModel(pl.LightningModule):
    def forward(self, x):
        return x[-1]
```
Of course the dimension needs to be the correct one (I disregarded the batch dim here).

Maybe we can even inherit from one of the `UNetBase`-Models.

The result of running this model here should of course be the same as the current implementation.

**Comment by rutkovskii** [comment]
1. The "current implementation" is the "persistence" section of the code for computing metrics?
2. The vision is to make `UNet`, `UNet Attention`, `UNetDS`, `SmaAt-UNet`, and `Persistence Model`? 

---

Then for the testing #9 and #28:
1. I extend `test_step()` for the `UNet_base` to include all metrics from `model` section of `method_test.py` (#40)
2. For `class PersistenceModel(Unet_base)` I implement the forward method to return the last image of the input.
3. Previous point means that `PersistenceModel` would inherit extended `test_step()` of the `Unet_base`. 

---
In a nutshell, the testing script would change from calling `get_metrics()`, to calling the `test_step()` that includes all metrics making the test script just a logic for making dataloader, running all models, and putting the results into the file. Correct?

Let me know if you have any difficulties understanding my questions.

**Comment by HansBambel** [comment]
To 1. and 2.: Yes, exactly.
---
To 1.: Hmm, I am not sure how well we can read out the computed metrics using the `test_step`-function. Since this only "logs" the metrics and I would like to have them saved to a json. Basically what is happening in `calc_metrics_test_set.py` in `get_metrics_from_model`. When we can utilize lightning here as well that would be great of course!
To 2.: Yes!
To 3.: Yes. I think we can also rename `Unet_Base` to `UnetBase` to follow proper coding practices :D

**Comment by rutkovskii** [comment]
Understood. I will proceed with it:
1. I think I will have to do the `PersistenseModel`.
2. Then switch to adding the metrics to `test_step` with metrics and making an analogous script as in calc_metrics_test_set.py

**Comment by rutkovskii** [comment]
After making `PersistenceModel` class and running Persistence from `test_precip_lightning.py` and the `test_step`. :

Here are the results:
```
{'Persistence': [{'MSE': 0.9015570878982544, 'MSE_denormalized': 2062.50146484375}], 
'Persistence_test': [{'MSE': 0.9015569686889648, 'MSE_denormalized': 2062.501220703125}]}
```
